diff --git a/scc/device_monitor.py b/scc/device_monitor.py
index 24b6942..f3f31f4 100644
--- a/scc/device_monitor.py
+++ b/scc/device_monitor.py
@@ -16,9 +16,12 @@ RE_BT_NUMBERS = re.compile(r"[0-9A-F]{4}:([0-9A-F]{4}):([0-9A-F]{4}).*")
 HCIGETCONNLIST = IOR(ord('H'), 212, ctypes.c_int)
 HAVE_BLUETOOTH_LIB = False
 try:
-	btlib_name = find_library('bluetooth')
-	assert btlib_name
-	btlib = ctypes.CDLL(btlib_name)
+	try:
+		btlib = ctypes.cdll.LoadLibrary("libudev.so")
+	except OSError:
+		btlib = ctypes.CDLL(find_library("bluetooth"))
+		if btlib is None:
+			raise ImportError("No library named bluetooth")
 	HAVE_BLUETOOTH_LIB = True
 except: pass
 

