diff --git a/scc/lib/eudevmonitor.py b/scc/lib/eudevmonitor.py
index 30dee24..ad32385 100644
--- a/scc/lib/eudevmonitor.py
+++ b/scc/lib/eudevmonitor.py
@@ -20,7 +20,6 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 """
 
 from collections import namedtuple
-from ctypes.util import find_library
 import os, ctypes, errno
 
 class Eudev:
@@ -28,8 +27,10 @@ class Eudev:
 	
 	def __init__(self):
 		self._ctx = None
-		assert find_library(self.LIB_NAME), self.LIB_NAME + "library not found"
-		self._lib = ctypes.CDLL(find_library(self.LIB_NAME))
+		try:
+			self._lib = ctypes.cdll.LoadLibrary("libudev.so")
+		except OSError:
+			raise ImportError("No library named udev")
 		Eudev._setup_lib(self._lib)
 		self._ctx = self._lib.udev_new()
 		if self._ctx is None:
